### ISSUE TITLE
softmax issue

### ISSUE THREAD

**Issue by HenryZ5734** [xianggkl/2023hitsz_cv#1]
My implementation of softmax.backward is below:
def backward(self, label):
        self.delta = self.softmax.copy()
        ### start your code
        self.delta -= label
        ## end your code
        return self.delta / self.input_batch

**Comment by xianggkl** [comment]
Thank you for pointing out my problem. In order to treat a batch of data as one, the cross-entropy function needs to take the average of the batches of data, so it also needs to divide by batchsize when back-propagating.
